### PR TITLE
Made some improvements

### PR DESCRIPTION
**Pull request by vnandwana**
* Upgraded killbill-oss-parent to the latest.
* Addressed all transitive dependency conflict issues.
* Implemented all methods that got added in the recent org.jooq.record* interfaces.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,107 +17,147 @@
...
+        <url>http://github.com/killbill/killbill-payment-test-plugin/tree/master</url>
+    </scm>
+    <issueManagement>
+        <system>Github</system>
+        <url>https://github.com/killbill/killbill-payment-test-plugin/issues</url>
+    </issueManagement>
     <properties>
+        <check.fail-spotbugs>true</check.fail-spotbugs>
+        <check.spotbugs-exclude-filter-file>spotbugs-exclude.xml</check.spotbugs-exclude-filter-file>
+        <maven.compiler.source>8</maven.compiler.source>
```
Comment (pierre) [inline pom.xml:43]: Hopefully we can remove this in the `work-for-release-0.23.x` branch?

File: pom.xml [hunk 2]
```diff
@@ -17,107 +17,147 @@
...
-            <type>test-jar</type>
-            <scope>test</scope>
+            <groupId>org.kill-bill.billing</groupId>
+            <artifactId>killbill-platform-osgi-bundles-lib-killbill</artifactId>
         </dependency>
         <dependency>
             <groupId>org.kill-bill.billing</groupId>
             <artifactId>killbill-platform-test</artifactId>
             <scope>test</scope>
+            <exclusions>
```
Comment (pierre) [inline pom.xml:111]: When doing the upgrade on `work-for-release-0.23.x`, can you check if all of these exclusions can be removed?

File: src/test/java/org/killbill/billing/plugin/payment/TestingStatesTest.java [hunk 3]
```diff
@@ -22,49 +22,49 @@
...
 public class TestingStatesTest {
 
-    TestingStates testingStates = new TestingStates();
+    private final TestingStates testingStates = new TestingStates();
 
     @Test
     void testAddKnownMethod() {
-        Assert.assertEquals(this.testingStates.add(TestingStates.Actions.ACTION_RETURN_PLUGIN_STATUS_CANCELED,
-                                                   "authorizePayment"), true);
+		Assert.assertTrue(this.testingStates.add(TestingStates.Actions.ACTION_RETURN_PLUGIN_STATUS_CANCELED,
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/payment/TestingStatesTest.java:29]: Whitespace seems off, can you reformat in IntelliJ?

File: src/main/java/org/killbill/billing/plugin/payment/dao/gen/tables/records/TestpaymentHppRequestsRecord.java [hunk 4]
```diff
@@ -408,7 +406,55 @@ public TestpaymentHppRequestsRecord values(ULong value1, String value2, String v
         return this;
     }
 
-    // -------------------------------------------------------------------------
+
+	@Override
+	public ULong component1() {
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/payment/dao/gen/tables/records/TestpaymentHppRequestsRecord.java:411]: Did you regenerate these files per the tool? Everything under the `gen` package is autogenerated.

Check https://github.com/killbill/killbill-stripe-plugin/blob/master/src/main/resources/README.txt for instructions.

File: src/test/java/org/killbill/billing/plugin/payment/TestingStatesTest.java (hunk 3 above)
Comment (vnandwana) [reply src/test/java/org/killbill/billing/plugin/payment/TestingStatesTest.java:29]: This looks correct in IntelliJ.

File: src/main/java/org/killbill/billing/plugin/payment/dao/gen/tables/records/TestpaymentHppRequestsRecord.java (hunk 4 above)
Comment (vnandwana) [reply src/main/java/org/killbill/billing/plugin/payment/dao/gen/tables/records/TestpaymentHppRequestsRecord.java]: I did now.

**Pull request merged**